src/main.rs

```diff
@@ -257,50 +286,51 @@ fn run_rust(context: &RunContext) -> Result<ExecResult> {
     })
 }
 fn run_cpp(context: &RunContext) -> Result<ExecResult> {
-    let tmp_dir = tempfile::tempdir()?;
-
-    let mut output_path = tmp_dir.path().to_path_buf();
-    output_path.push("sol");
-
-    let mut res = Command::new("/usr/bin/g++")
-        .args(vec![
-            "-Wall",
-            "--std=c++17",
-            "-O3",
-            context.abs_solution().to_str().unwrap(),
-            "-o",
-            output_path.to_str().unwrap(),
-        ])
-        .spawn()?;
+    let mut times = vec![];
+    for i in 0..context.times {
+        let tmp_dir = tempfile::tempdir()?;
+
+        let mut output_path = tmp_dir.path().to_path_buf();
+        output_path.push("sol");
+
+        let mut res = Command::new("/usr/bin/g++")
+            .args(vec![
+                "-Wall",
+                "--std=c++17",
+                "-O3",
+                context.abs_solution().to_str().unwrap(),
+                "-o",
+                output_path.to_str().unwrap(),
+            ])
+            .spawn()?;
```

Comment:
Why recompile the solution with every run?

Reply:
The commit message actually explains! I tried doing a staging/tmp dir separation, but ran into borrow checker issues. I know I need more than 10 minutes to understand exactly what value was being moved and I didn't have the time.

Reply:
@chermehdi  By the way. I don't post a PR to contribute the change. I post it to provide the feedback you asked for. In this particular topic, did you notice the problem? On second run people can basically just have `main() {}` and everything will pass in 1ms. Literally. I tried.

Reply:
none of this will pass the actual final testing, people can do any hacks they need as long as the solution will provide a proper output for a proper input :)

Reply:
Okay then. It's a bit of a shame that everything was so transparent, but then, magically "the final testing will be different". But good to hear none-the-less. That was actually the point. I'll take it it will run on a consistent platform as well, so that results are fair.

Cheers

Reply:
@sehe I don't know what you understood from my comment above, but let me clarify: Everything that will be part of the final testing will still be open, the hardware used, the strategy and any changes to runner will be published here in this repo, feel free to check further changes as they come along.

Cheers.

Reply:
Thanks for that clarification. It wasn't given, and the context in which you are saying it seems evasive - perhaps you meant to say you had already noticed this loophole and were working on fixes independently.

Also, I would note that there is a distinct difference between the judgement mechanism being open **before** closing (or, really, opening) submissions, and **after**. But I agree that the transparency is still very commendable, and it gives a solid ground for GeeksBlabla to motivate their change(s) to the effective judgement criteria. I will keep an eye on this repository. Best of luck!